### PR TITLE
Adding new function sg_params_checker to check acquisition parameters; Include it in CI

### LINKED ISSUE #113: Acquisition parameters QC
In the original version of the manuscript we had this paragraph and suggestion.

```
To make sure the protocol was followed without modifying the sequence parameters, the protocol used by each site was exported in pdf and sent to the central site for QCAing.

that's old school-- we should make a BIDS-friendly script to verify the parameters. 
```

@jcohenadad I started to work on a script using pybids. I would like to know on what parameters should we focus for an initial iteration. I thought of checking TR,TE,FA.

The strategy is to query the dataset and grab each contrast's metadata. This metadata will be compared to the template stored in a json file with specific sections for scanner, software version, contrast, TR,TE,FA. If they don't match a warning message will be triggered.

### PR DESCRIPTION
**Pull request by alexfoias**
This PR introduces parameters checking during continuous integration (CI), via Travis.

The CLI is also introduced: `sg_params_checker`.

### How to test this PR
~~~
git checkout -b af/acq_param_ci origin/af/acq_param_ci
pip install -e .
sg_params_checker -path-in <PATH>/data-single-subject/
~~~

Here are some results for current configuration.

https://travis-ci.com/github/spine-generic/data-single-subject/jobs/360645135#L275

Fixes #113

### CODE REVIEW

**Comment by alexfoias** [comment]
Here are the results from the latest version of the implementation: https://travis-ci.com/github/spine-generic/data-single-subject

**Comment by alexfoias** [comment]
@jcohenadad This PR is ready for review. I'll first nee to merge this one and then I can implement CI in the single & multi subject datasets.

File: spinegeneric/cli/params_checker.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    #Print WARNING log
+    if path_warning_log :
+        file = open(path_warning_log, 'r')
+        lines = file.read().splitlines()
+        file.close()
+        for line in lines:
+            print(line)
+
+if __name__ == '__main__':
+    main()
```
Comment (jcohenadad) [inline spinegeneric/cli/params_checker.py:82]: EOF

File: spinegeneric/cli/params_checker.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+from bids.tests import get_test_data_path
+import os
+import json
+import logging
+import argparse
+from pkg_resources import resource_filename
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        description="Acquistion parameters checker",
```
Comment (jcohenadad) [inline spinegeneric/cli/params_checker.py:14]: add more info about how this is done.

**Comment by jcohenadad** [comment]
this should pass:
~~~
WARNING: Incorrect RepetitionTime: sub-chibaIngenia_T2star.nii.gz; TR=0.625 instead of 0.6
WARNING: Incorrect EchoTime: sub-douglas_T2w.nii.gz; TE=0.12 instead of 0.118
WARNING: Incorrect RepetitionTime: sub-glen_T2star.nii.gz; TR=0.625 instead of 0.6
~~~

**Comment by alexfoias** [comment]
I updated the limits and addressed the other comments. 

As for the limits here is the output:

```
(base) alfoi@MacBook-Pro:spine-generic % sg_params_checker -path-in /Users/alfoi/code/data-single-subject/
/Users/alfoi/miniconda3/lib/python3.7/site-packages/bids/layout/models.py:102: FutureWarning: The 'extension' entity currently excludes the leading dot ('.'). As of version 0.14.0, it will include the leading dot. To suppress this warning and include the leading dot, use `bids.config.set_option('extension_initial_dot', True)`.
  FutureWarning)
WARNING: Incorrect FlipAngle: sub-douglas_T2w.nii.gz; FA=120 instead of 180
WARNING: Incorrect RepetitionTime: sub-mgh_T2w.nii.gz; TR=2 instead of 1.5
WARNING: Incorrect FlipAngle: sub-tokyoSigna1_T2star.nii.gz; FA=20 instead of 30
WARNING: Incorrect FlipAngle: sub-tokyoSigna2_T2star.nii.gz; FA=20 instead of 30
WARNING:sub-ucl_T2star.nii.gz Missing Manufacturer in json sidecar; Cannot check parameters.
```

File: spinegeneric/cli/params_checker.py [hunk 3]
```diff
@@ -1,3 +1,11 @@
+#!/usr/bin/env python
+#
+# Script to check parameters checker
```
Comment (jcohenadad) [inline spinegeneric/cli/params_checker.py:3]:
```suggestion
# Script to check acquisition parameters.
```

File: spinegeneric/cli/params_checker.py [hunk 4]
```diff
@@ -1,3 +1,11 @@
...
+#
+# For usage, type: sg_params_checker -path-in PATH_TO_BIDSDATASET
```
Comment (jcohenadad) [inline spinegeneric/cli/params_checker.py:5]:
```suggestion
# For usage, type: sg_params_checker -h
```
less maintenance

File: spinegeneric/cli/params_checker.py [hunk 5]
```diff
@@ -11,7 +19,9 @@
 
 def get_parser():
     parser = argparse.ArgumentParser(
-        description="Acquistion parameters checker",
+        description="Acquistion parameters checker feature. This feature allows the users"
+                    "to compare the acquisition parameters that can be found in the json "
+                    "sidecar to the recommended acquisition parameters."
```
Comment (jcohenadad) [inline spinegeneric/cli/params_checker.py]:
```suggestion
        description="This function checks if the image acquisition parameters of each "
                    "image (found in the json file) in a dataset conform to the required parameters "
                    "(listed for each vendor in the file cli/specs.json)."
```

File: spinegeneric/cli/params_checker.py [hunk 6]
```diff
@@ -80,4 +90,3 @@ def main():
 
 if __name__ == '__main__':
     main()
-
```
Comment (jcohenadad) [inline spinegeneric/cli/params_checker.py]: EOF

**Comment by alexfoias** [comment]
@jcohenadad Here is the output of the new version:
```
(base) alfoi@MacBook-Pro:~ % sg_params_checker -path-in /Users/alfoi/code/data-single-subject/
/Users/alfoi/miniconda3/lib/python3.7/site-packages/bids/layout/models.py:102: FutureWarning: The 'extension' entity currently excludes the leading dot ('.'). As of version 0.14.0, it will include the leading dot. To suppress this warning and include the leading dot, use `bids.config.set_option('extension_initial_dot', True)`.
  FutureWarning)
WARNING: Incorrect FlipAngle: sub-douglas_T2w.nii.gz; FA=120 instead of 180
WARNING: Incorrect RepetitionTime: sub-mgh_T2w.nii.gz; TR=2 instead of 1.5
WARNING: Incorrect FlipAngle: sub-tokyoSigna1_T2star.nii.gz; FA=20 instead of 30
WARNING: Incorrect FlipAngle: sub-tokyoSigna2_T2star.nii.gz; FA=20 instead of 30
WARNING:sub-ucl_T2star.nii.gz Missing Manufacturer in json sidecar; Cannot check parameters.
```

**Pull request merged**